### ISSUE TITLE
ObservableArray showing up as  empty object in Jest

### ISSUE THREAD

**Issue by WilfredKesh** [mobxjs/mobx-react#577]
I love using  mobx but running into some pitfalls during testing using Jest. I am assigning an array to a property in my store but when I try to test the property Jest is throwing an error: "Expected array but received an object."  this is what is being console logged.    
    ObservableArray [Array] {}

How do I convert this observable array into a regular array in my test.

tried this in my test file but its not working
    import {toJS} from 'mobx';
     expect(toJS(store.arrayProperty)).toEqual(testArray);

**Comment by mattruby** [comment]
Try
.slice

**Comment by WilfredKesh** [comment]
Thanks for responding . Unfortunately that was the first thing I tried.

`expect(store.agreements.slice()).toEqual(providerAgreements);`

but it did not work. I could be wrong but I think the array is passed on to Jest as an empty object so calling .slice() on an empty object will not work. The conversion may need to happen before its passed on to Jest. Calling .slice() on the array before its assigned to an observable property does not work either. So I am assuming you are suggesting to create a copy of the array and store it as a separate variable that is not an observable then test that non-observable variable?
```
```

export default class myStore {
  constructor(state = {}) {
    extendObservable(
      this,
      {
        observableArray: [],
.....
}
//end of constructor

copyArrayOfObservableArrayToBeTested = [];

  @action
  loadArray(array) {
this.copyArrayOfObservableArrayToBeTested = array.slice()
    this.observableArray = array;
  }
```


```

If this is what you are suggesting, it seems there should be cleaner way of testing the actual store property instead of testing a copy of the store property

**Comment by mattruby** [comment]
Hard to make complete sense of things on my phone. Can you create a small
codesandbox.io example? It has great testing support.

**Comment by mweststrate** [comment]
@WilfredKesh yes, please create a small, cohesive snippet or sandbox. Your first comment is testing totally different things as you are introducing in your second one, so it is kinda impossible to get an idea what you are doing

**Comment by mweststrate** [comment]
@WilfredKesh looks fine at first sight, but there are no known bugs in this area, could you please setup a reproduction codesandbox.io? It has Jest support built in.
